docs/plugins/creating-plugin.rst

```diff
@@ -64,11 +68,82 @@ to ensure your plugin's extraordinary quality.
+ * Bundle class must use the ``Sylius\Bundle\CoreBundle\Application\SyliusPluginTrait`` trait.
+ * Namespace should follow _`PSR-4 <http://www.php-fig.org/psr/psr-4/>`. It should start with vendor name, followed by ``Sylius`` and suffixed by ``Plugin`` (e.g. ``VendorName\SyliusInvoicePlugin``)
+
+.. note::
+
+    Following the naming strategy for the bundle class & extension class prevents configuration key collision. Following the convention mentioned
+    above generates the default configuration key as e.g. ``vendor_name_sylius_invoice_plugin``.
+
+The rules are to be applied to all bundles which will provide an integration with the whole Sylius platform
+(``sylius/sylius`` or ``sylius/core-bundle`` as dependency).
+
+Reusable components for the whole Symfony community, which will be based just on some Sylius bundles should follow
+the regular Symfony conventions.
+
+Example
+~~~~~~~
+
+Assuming you are creating the invoicing plugin as used above, this will result in the following set-up.
+
+**1.** Name your repository: ``vendor-name/sylius-invoice-plugin``.
+
+**2.** Create bundle class in ``src/VendorNameSyliusInvoicePlugin.php``:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace VendorName\SyliusInvoicePlugin;
+
+    use Sylius\Bundle\CoreBundle\Application\SyliusPluginTrait;
+    use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+    final class AcmeExamplePlugin extends Bundle
+    {
+        use SyliusPluginTrait;
+    }
+
+**3.** Create extension class in ``src/DependencyInjection/VendorNameSyliusInvoiceExtension.php``:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace VendorName\SyliusInvoicePlugin\DependencyInjection;
+
+    use Symfony\Component\Config\FileLocator;
+    use Symfony\Component\DependencyInjection\ContainerBuilder;
+    use Symfony\Component\DependencyInjection\Extension\Extension;
+    use Symfony\Component\DependencyInjection\Loader\XmlFileLoader;
+
+    final class VendorNameSyliusInvoiceExtension extends Extension
+    {
+        /**
+         * {@inheritdoc}
+         */
+        public function load(array $config, ContainerBuilder $container): void
+        {
+            $config = $this->processConfiguration($this->getConfiguration([], $container), $config);
+            $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
+        }
+    }
+
+**4.** In ``composer.json``, define the correct namespacing for the PSR-4 autoloader:
+
+.. code-block:: json
 
-The following rules are applied to all bundles which will provide an integration with the whole Sylius platform
-(``sylius/sylius`` or ``sylius/core-bundle`` in vendors). Reusable components for the whole Symfony community, which will be based
-just on some Sylius bundles should follow regular Symfony conventions.
+    "autoload": {
```

Comment:
The build is failing because the given JSON snippet is invalid, let's change it to:

```json
{
  "autoload": {
    "psr-4": {
      "VendorName\\SyliusInvoicePlugin\\": "src/",
      "Tests\\VendorName\\SyliusInvoicePlugin\\": "tests/"
    }
  }
}